crates/gitbutler-core/src/ops/entry.rs

```diff
@@ -3,30 +3,33 @@ use anyhow::Result;
 use itertools::Itertools;
 use serde::Deserialize;
 use std::fmt;
-use std::fmt::Display;
 use std::fmt::Formatter;
+use std::fmt::{Debug, Display};
 use std::path::PathBuf;
 use std::str::FromStr;
 use strum::EnumString;
 
+use crate::git;
 use serde::Serialize;
 
 /// A snapshot of the repository and virtual branches state that GitButler can restore to.
 /// It captures the state of the working directory, virtual branches and commits.
 #[derive(Debug, PartialEq, Clone, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Snapshot {
-    /// The sha of the commit that represents the snapshot
-    pub id: String,
-    /// Snapshot creation time in epoch seconds
-    pub created_at: i64,
+    /// The id of the commit that represents the snapshot
+    #[serde(rename = "id", with = "crate::serde::oid")]
+    pub commit_id: git::Oid,
```

Comment:
oh, never mind, i missed the rename

Reply:
Admittedly, I should probably have renamed the UI part instead of doing it in the (for me) simpler way.
Maybe the UI also benefits from the clearer name.